### PR TITLE
[demo] -Set flagd memory limit to 50Mi

### PR DESCRIPTION
**Pull request by puckpuck**
This replaces #1131 and fixes [issue 1536](https://github.com/open-telemetry/opentelemetry-demo/issues/1536) in the demo repo.

Increases the memory limit for FlagD to be 50Mi instead of 20Mi

This also bumps the Flagd version the latest release (0.10.1)

### CODE REVIEW

File: charts/opentelemetry-demo/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 type: application
 name: opentelemetry-demo
-version: 0.30.3
+version: 0.30.5
```
Comment (jaronoff97) [inline charts/opentelemetry-demo/Chart.yaml:4]: this should be `0.30.4`

File: charts/opentelemetry-demo/values.yaml [hunk 2]
```diff
@@ -527,15 +527,15 @@ components:
     enabled: true
     imageOverride:
       repository: "ghcr.io/open-feature/flagd"
-      tag: "v0.9.0"
+      tag: "v0.10.1"
```
Comment (jaronoff97) [inline charts/opentelemetry-demo/values.yaml:530]: can you mention the tag bump in the description?

File: charts/opentelemetry-demo/Chart.yaml (hunk 1 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/Chart.yaml]: I set it to 0.30.5 in anticipation of #1156 getting merged.

File: charts/opentelemetry-demo/values.yaml (hunk 2 above)
Comment (puckpuck) [reply charts/opentelemetry-demo/values.yaml:534]: Done

**Review by JaredTan95 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**